kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -246,7 +248,25 @@ public void put(Collection<SinkRecord> records) {
 
     for (SinkRecord record : records) {
       if (record.value() != null || config.getBoolean(BigQuerySinkConfig.DELETE_ENABLED_CONFIG)) {
-        PartitionedTableId table = getRecordTable(record);
+        PartitionedTableId table;
+        if(enableRetries) {
+          try {
+            table = getRecordTable(record);
+          } catch (RetriableException e) {
+            if(remainingRetries == 0) {
+              throw new ConnectException(e);
+            } else {
+              logger.warn("Write of records failed, remainingRetries={}", remainingRetries);
+              remainingRetries--;
+              throw e;
+            }
+          } catch (Exception e) {
```

Comment:
This catch block should not be required. If it is not retriable , we are only catching and re-throwing.

Reply:
Okay sure, done